File: sdk/src/test/java/com/cxense/cxensesdk/UserAgentInterceptorTest.java
package com.cxense.cxensesdk;

import org.junit.Test;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.mockwebserver.MockResponse;
import okhttp3.mockwebserver.RecordedRequest;

import static org.junit.Assert.assertEquals;
import static org.powermock.api.mockito.PowerMockito.doReturn;
import static org.powermock.api.mockito.PowerMockito.mock;

/**
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-06-05).
 */
public class UserAgentInterceptorTest extends BaseInterceptorTest {
    private static final String USER_AGENT = "some custom test user agent";

    @Test
    public void intercept() throws Exception {
        mockWebServer.enqueue(new MockResponse());
        UserAgentProvider mockUserAgentProvider = mock(UserAgentProvider.class);
        doReturn(USER_AGENT).when(mockUserAgentProvider).getUserAgent();

        OkHttpClient okHttpClient = new OkHttpClient()
                .newBuilder()
                .addInterceptor(new UserAgentInterceptor(mockUserAgentProvider))
                .build();
        okHttpClient.newCall(new Request.Builder().url(mockWebServer.url("/")).build()).execute();

        RecordedRequest request = mockWebServer.takeRequest();
        assertEquals(USER_AGENT, request.getHeader("User-Agent"));
    }

}

File: sdk/src/test/java/com/cxense/cxensesdk/UserAgentInterceptorTest.kt
package com.cxense.cxensesdk

import com.nhaarman.mockitokotlin2.doReturn
import com.nhaarman.mockitokotlin2.mock
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.mockwebserver.MockResponse
import kotlin.test.Test
import kotlin.test.assertEquals

class UserAgentInterceptorTest : BaseInterceptorTest() {
    private val userAgentProvider: UserAgentProvider = mock {
        on { userAgent } doReturn USER_AGENT_VALUE
    }
    private val userAgentInterceptor = UserAgentInterceptor(userAgentProvider)

    @Test
    fun intercept() {
        mockWebServer.enqueue(MockResponse())
        OkHttpClient.Builder()
            .addInterceptor(userAgentInterceptor)
            .build()
            .newCall(
                Request.Builder()
                    .url(mockWebServer.url("/"))
                    .build()
            ).execute()
        assertEquals(USER_AGENT_VALUE, mockWebServer.takeRequest().getHeader("User-Agent"))
    }

    companion object {
        private const val USER_AGENT_VALUE = "test user agent"
    }
}
